Add tests for color.grayscale()

dvdherron opened the pull request:
![](https://source.unsplash.com/featured/?cute,animal&grey)
<!-- Swap `CHANGE_ME` for a random string to get a different image -->


<!--
## Description
Uncomment if you want to provide a custom description.
-->

dvdherron committed:
first draft of updated grayscale specs

dvdherron committed:
add more non legacy tests

dvdherron commented on spec/core_functions/color/grayscale.hrx:
```diff
@@ -74,7 +74,49 @@ a {b: grayscale($color: white)}
 <===> named/output.css
 a {
-  b: white;
+  b:  white;
```
@mirisuzanne 

For legacy colors, is it implied that the [conversion happens in `hsl`](https://github.com/sass/sass/blob/colors-2/proposal/color-4-new-spaces.md#colorgrayscale) but the output color is always returned in the origin space?

mirisuzanne replied:
With legacy colors, we don't guarantee that the output will be in a particular space. Generally the goal is that most existing functions will continue to work for legacy colors exactly the way they did before. So whatever grayscale currently returns with a legacy color is probably what it will keep returning.

mirisuzanne replied:
I think I might have been wrong about this, and we are planning to maintain the initial color space - which is rgb for hex and named colors. Might be worth confirming that with Natalie, though?

nex3 committed:
[Color 4] Update tests for Color Level 4 (#1846)

This updates most tests for the currently-implemented chunk of the
Color 4 feature. A few tests that need more thorough reworking (such
as being moved out of `error/` for things that are no longer intended
to be errors) are marked as "todo" instead.

See #1828
See sass/sass#2831

dvdherron committed:
Merge branch 'feature.color-4' into grayscale

dvdherron committed:
Merge branch 'main' into grayscale

dvdherron committed:
update output

dvdherron committed:
Update output expectations to match Dart Sass

nex3 committed:
Remove redundant libsass output

nex3 committed:
Poke CI
